src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs

```diff
@@ -56,11 +56,33 @@ public static ICollection<ISymbol> GetMissingEnumMembers(ISwitchExpressionOperat
 
                     enumMembers.Remove(IntegerUtilities.ToInt64(constantValue.Value));
                 }
+                else if (arm.Pattern is IBinaryPatternOperation binaryPattern)
+                {
+                    HandleBinaryPattern(binaryPattern, enumMembers);
+                }
```

Comment:
1. do we need a 'false' here to handle all other types of pattern?
2. should HandleBinaryPattern return a bool to say if it couldn't handle this

It feels like we want this so we can appropriately bail out of the feature on things taht we don't understand.

Reply:
@CyrusNajmabadi What are other patterns that can be relevant for enums?

I don't think we should bail out for things we don't understand. For example, if an arm has an "and" pattern, we still need to analyze other arms. See `TestMixingOrWithAndPatterns2`, for which I think the current behavior is correct.

Reply:
good point.  thanks.  but then why does the higher 'if' bail out on a value it doesn't understand.  feels like we shouidl at least be conssitent on that point wrt constant patterns.

Reply:
@CyrusNajmabadi Agreed. I updated it to be consistent.